pkg/asset/manifests/openstack/cluster.go

```diff
@@ -24,6 +25,12 @@ const (
 func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) (*capiutils.GenerateClusterAssetsOutput, error) {
 	manifests := []*asset.RuntimeFile{}
 	openstackInstallConfig := installConfig.Config.OpenStack
+
+	var externalNetwork *capo.NetworkParam
+	if e := openstackInstallConfig.ExternalNetwork; e != "" {
+		externalNetwork = &capo.NetworkParam{Filter: &capo.NetworkFilter{Name: e}}
+	}
```

Comment:
and if it's empty we should probably set `DisableExternalNetwork`, otherwise CAPO will look for any "External" network
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/docs/book/src/topics/crd-changes/v1alpha7-to-v1beta1.md?plain=1#L307